core/src/main/java/com/graphhopper/util/shapes/Polygon.java

```diff
@@ -192,4 +221,21 @@ public double getMaxLat() {
     public double getMaxLon() {
         return maxLon;
     }
+
+    public static Polygon parsePoints(String pointsStr) {
+        String[] arr = pointsStr.split(",");
+
+        double[] lats = new double[arr.length /2];
+        double[] lons = new double[arr.length /2];
+
+        for (int j = 0; j < arr.length; j++) {
+            if (j % 2 == 0) {
+                lats[j / 2] = Double.parseDouble(arr[j]);
+            } else {
+                lons[(j - 1) / 2] = Double.parseDouble(arr[j]);
+            }
+        }
+
+        return new Polygon(lats, lons);
```

Comment:
@boldtrn do you think "lat,lon,lat2,lon2,..." is a good format? At least we should check that the count is dividable by 2

Reply:
Mhm, good question. My issue with this is that it is really hard to read and write for humans. It is easy for computers though. We should definitely check if the length is a multiple of 2. Maybe a geojson like format would be better? e.g. [[lon,lat],[lon2,lat2]]... But then we don't do this for BBox. Maybe we should do this for BBox and Polygon?

Reply:
When I think about this, the only format that makes sense would be like it is now. But this shows again the ugliness of the URL request approach that we currently have and we should invest more into 1112 to being able to request via JSON where we then could easily use GeoJSON for something like this.